File: src/RegExMatch.java
package org.nlogo.extensions.string;

import org.nlogo.api.Argument;
import org.nlogo.api.Context;
import org.nlogo.api.DefaultReporter;
import org.nlogo.api.ExtensionException;
import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;

public class RegExMatch  extends DefaultReporter
{   
    // rex-match  needle  haystack // reports true if the regular expression matches the string
    
    public Syntax getSyntax()
    {
        return Syntax.reporterSyntax(
            new int[] { Syntax.StringType(), Syntax.StringType() }, Syntax.BooleanType()
        ) ;
    }
    
    public Object report(Argument args[], Context context)
        throws ExtensionException, LogoException
    {
        // use typesafe helper method from 
        // org.nlogo.api.Argument to access argument
        String regexNeedle = args[0].getString() ;
        String hayStack = args[1].getString() ;
        
        Boolean result = new Boolean (hayStack.matches(regexNeedle) ) ;
        
        return result ;

    }
}

File: src/RegExMatch.scala
package org.nlogo.extensions.string

import org.nlogo.api.Argument
import org.nlogo.api.Context
import org.nlogo.api.Reporter
import org.nlogo.api.ExtensionException
import org.nlogo.api.LogoException
import org.nlogo.core.Syntax

class RegExMatch extends Reporter
{
    // rex-match  needle  haystack // reports true if the regular expression matches the string
    def getSyntax(): Syntax =
    {
      Syntax.reporterSyntax(
        right = List(Syntax.StringType, Syntax.StringType),
        ret   = Syntax.BooleanType)
    }

    @throws(classOf[ExtensionException])
    @throws(classOf[LogoException])
    def report(args: Array[Argument], context: Context): Object =
    {
      val regexNeedle: String = args(0).getString
      val hayStack: String = args(1).getString
      val result = hayStack.matches(regexNeedle)

      result.asInstanceOf[AnyRef]
    }
}
